lib/telemetry_metrics.ex

```diff
@@ -563,6 +629,25 @@ defmodule Telemetry.Metrics do
           "expected metric name to be a string or a list of atoms, got #{inspect(term)}"
   end
 
+  defp validate_event_filter!(nil), do: nil
+  defp validate_event_filter!(filter) when is_function(filter, 1), do: filter
+
+  defp validate_event_filter!(term) do
+    raise ArgumentError,
+          "expected filter to be a function accepting metadata, got #{inspect(term)}"
+  end
+
+  def validate_event_filter_options!(keep, drop) when is_nil(keep) or is_nil(drop), do: :ok
+
+  def validate_event_filter_options!(_keep, _drop) do
+    raise ArgumentError,
+          "events can be filtered with the keep or drop option, but not both"
+  end
+
+  def filter(nil, nil), do: nil
+  def filter(nil, drop), do: drop
+  def filter(keep, nil), do: fn meta -> not keep.(meta) end
```

Comment:
The semantic here is still confusing, sorry. `filter` in Elixi means to `keep` and here it means `not keep`. What if we simply call the field `keep` to avoid any confusion? Then we remove any occurrence of `filter` from the code to avoid confusion.